docs/yaml/functions/response_file.yaml

```diff
@@ -0,0 +1,71 @@
+name: response_file
+returns: list[str] | str
+description: |
+  Write argument list to a response file, if the resulting
+  length of the command line is longer than a maximum value.
+
+  Returns either the list of arguments or the path to the response
+  file.
+
+  This function is designed to be used with `custom_target`,
+  `run_target`, `run_command`, `test`, or any other function
+  running a custom executable or script.
+
+varargs:
+  name: arguments
+  type: str | file | list
+  description: The argument list.
+
+kwargs:
+  prefix:
+    type: str
+    default: "'@'"
+    description: |
+      The prefix to prepend to the response file path.
+      Use an empty string if you don't need a prefix.
+  
+  quote:
+    type: str | bool
+    default: true
+    description: |
+      If `true`, quote the arguments in the response file
+      using the default quote function, based on the operating
+      system.
+
+      If `false`, do not quote the arguments.
+
+      Can also be `'sh'` to use sh-like quoting,
+      `'cmd'` for Windows cmd-like quoting,
+      or `'gcc'` for gcc-like quoting.
+
+  max_args_length:
+    type: int | bool
+    default: 127
+    description: |
+      Maximum allowed command line length, after joining and
+      quoting the arguments. It doesn't take into account the
+      lenght of the command itself, or the length of other
+      arguments that could be provided to the comamnd.
+
+      If `true` or `0`, a response file is always created,
+      no mater the length of the command line.
+
+      If `false`, the function do not creates a response file,
+      but returns the list of arguments.
+  
+  separator:
+    type: str
+    default: "' '"
+    description: |
+      The separator used to join the arguments. Usually a
+      space character, but could also be a `'\n'` to put
+      one argument per line in the response file.
+
+  relative_paths:
+    type: bool
+    default: true
+    description: |
+      Whether to use relative or absolute paths for files
+      in the response file. This only applies to `file`
```

Comment:
I think there is a way to link to the actual file type. `[[file]]` maybe

Reply:
I would specify relative to what, top builddir I guess.

Reply:
That has been requested in the past.

Reply:
> do we even need relative path at all?

In my `xgettext` usecase, it is better to have relative paths, otherwise .pot files contain paths from my machine. But I agree absolute paths should be the default.

> That has been requested in the past.

How could something not existing yet could be requested in the past?

Reply:
> Surprisingly custom_target() and run_target() does not have a workdir kwarg, but that could be added at some point.

^this